src/pds/updart/client.py

```diff
@@ -62,39 +66,138 @@ def has_target(self, identifier: str):
         self.__add_clause(clause)
         return self
 
+    def has_investigation(self, identifier: str):
+        """Selects products having a given investigation.
+
+        Lazy evaluation is used to only apply the filter when one iterates on it.
+        This is done so that multiple filters can be combined before the request is actually sent.
+        :param identifier: lidvid of the target
+        :return: a Products instance with the investigation filter added.
+        """
+        clause = f'ref_lid_investigation eq "{identifier}"'
+        self.__add_clause(clause)
+        return self
+
     def before(self, d: datetime):
-        """Selects products which start date is before given datetime.
+        """Selects products with a start date before the given datetime.
 
         :param d: datetime
-        :return: a Products instance with before filter applied
+        :return: a Products instance with a before filter applied
         """
         iso8601_datetime = d.isoformat().replace("+00:00", "Z")
         clause = f'pds:Time_Coordinates.pds:start_date_time le "{iso8601_datetime}"'
         self.__add_clause(clause)
         return self
 
     def after(self, d: datetime):
-        """Selects products which end date is after given datetime.
+        """Selects products with an end date after the given datetime.
 
         :param d: datetime
-        :return: a Products instance with after filter applied
+        :return: a Products instance with an after filter applied
         """
         iso8601_datetime = d.isoformat().replace("+00:00", "Z")
         clause = f'pds:Time_Coordinates.pds:stop_date_time ge "{iso8601_datetime}"'
         self.__add_clause(clause)
         return self
 
-    def of_collection(self, id: str):
-        """Selects products which belong to a collection.
+    def of_collection(self, identifier: str):
+        """Selects products which belong to a Collection.
+
+        :param identifier: the collection id, e.g. a lidvid
+        :return: a Products instance with a parent collection identifier filter applied
+        """
+        clause = f'ops:Provenance.ops:parent_collection_identifier eq "{identifier}"'
+        self.__add_clause(clause)
+        return self
+
+    def observationals(self):
+        """Selects Observational products for a specific filter.
+
+        :return: a Products instance with Observational product class filter applied
+        """
+        clause = 'product_class eq "Product_Observational"'
+        self.__add_clause(clause)
+        return self
+
+    def collections(self, type: Optional[str] = None):
+        """Selects Collection products for a specific filter.
```

Comment:
Not a critical feedback; I don't understand what the meaning of "for a specific filter" in this context.
I would say "Selects products of class Collection only" or "Select Collections only".